src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -318,38 +318,25 @@ std::vector<DSAInfo> Flasher::getInstalledDSA()
 DSAInfo Flasher::getOnBoardDSA()
 {
     std::string vbnv;
-    std::string bmc;
     uint64_t ts = NULL_TIMESTAMP;
-
+    std::string bmc;
     std::string board_name;
     std::string uuid;
     bool is_mfg = false;
 
     is_mfg = xrt_core::query_device<bool>(m_device, xrt_core::device::QR_IS_MFG);
-    board_name = xrt_core::query_device<std::string>(m_device, xrt_core::device::QR_ROM_FPGA_NAME);
-    uuid = xrt_core::query_device<std::string>(m_device, xrt_core::device::QR_ROM_UUID);
-
+    board_name = xrt_core::query_device<std::string>(m_device, xrt_core::device::QR_BOARD_NAME);
     
-    if (is_mfg)
-    {
+    if (is_mfg) {
         std::stringstream ss;
-
         ss << "xilinx_" << board_name << "_GOLDEN_" << mGoldenVer;
         vbnv = ss.str();
-    }
-    //else if (mFRHeader.VBNVName[0] != '\0')
-    //{
-    //    vbnv = std::string(reinterpret_cast<char *>(mFRHeader.VBNVName));
-    //    ts = mFRHeader.TimeSinceEpoch;
-    //}
-    else if (uuid != xrt_core::invalid_query_value<std::string>())
-    {
+    } else {
         vbnv = xrt_core::query_device<std::string>(m_device, xrt_core::device::QR_ROM_VBNV);
         ts = xrt_core::query_device<uint64_t>(m_device, xrt_core::device::QR_ROM_TIME_SINCE_EPOCH);
-    }
-    else
-    {
-        std::cout << "ERROR: Platform name not found" << std::endl;
+        uuid = xrt_core::query_device<std::string>(m_device, xrt_core::device::QR_ROM_UUID);
+        if (vbnv.empty() || ts == std::numeric_limits<uint64_t>::max())
```

Comment:
1) Could you break this error condition into two separate errors (one for invalidate timestamp and the other missing vbnv value.

2) Since an error is generated, an exception should be thrown and caught by the calling method.

Reply:
Yes. Will do that